### PR TITLE
Improve list removal behavior

### PR DESCRIPTION
**Pull request by abeforgit**
Vastly improved list removal logic, now properly handles nested lists
Probably still missing edge cases, but I think this is in the "usually does what it should, and otherwise doesnt blow up" stage

But please track any issues you find with list removal in the comments here

### CODE REVIEW

**Comment by abeforgit** [comment]
fixes https://binnenland.atlassian.net/browse/GN-2152

**Comment by nvdk** [comment]
Did some testing and this is definitely an improvement! I'm just testing list removal as this is the main concern for this PR. A general remark is that the pattern "remove list" -> "reapply list" -> "remove list" is pretty broken for me. Currently testing in firefox.

Now on to the issues I still encounter:

given the list of `case 20` under lists in the dummy app:
* select part the second list item, all of it's sublist items and the final list item. 
remove list will not remove the item from the top list, this was unexpected to me, but may be by design. at least it behaves very consistently and I could argue this to be expected behaviour :).

* putting the cursor on  "subitem 1" and pressing remove list
results in undesirable behaviour for sub item 2, which I don't think would be resolved by merging sibling lists.

File: addon/components/rdfa/editor-toolbar.hbs [hunk 1]
```diff
@@ -64,7 +64,7 @@
       </button>
       {{/if}}
       {{#if @showIndentButtons}}
-      <button type="button" class="say-toolbar__button" {{on "click" this.insertUnindent}} data-test-button-id="insert-unindent">
+      <button type="button" class="say-toolbar__button {{if this.isInList "is-active"}}" {{on "click" this.insertUnindent}} data-test-button-id="insert-unindent">
```
Comment (nvdk) [inline addon/components/rdfa/editor-toolbar.hbs:67]: This seems counter intuitive, the button is not active when in a list. I think you want `class={{unless this.isInList "disabled"}}`, as well as a disabled state on the button itself

Comment (abeforgit) [reply addon/components/rdfa/editor-toolbar.hbs:67]: I think this is changed in the indent/unindent prs, so I'd leave it as is and see how it behaves after all merges

**Comment by Asergey91** [comment]
not sure that I have anything meaningfull to add to be honest.
cases like these seem to be wierd
when removing list item 6 
it seems to unindent item 7 to the most parent ul
this seemed strange to me but then i checked on ck editor and it seems to do the same.
Screenshot from 2021-02-08 15-00-35

**Comment by abeforgit** [comment]
thanks for trying that out, not a case I tested. But yeah the way it handles it now seems sensible, I don't think there is a "natural" behavior for this case which everyone can agree on.

**Comment by abeforgit** [comment]
@nvdk I think all the cases you mentioned are handled now, and it's now adding br's where needed much more consistently. I think it's ready to be merged

**Review by nvdk — APPROVED** [review]

**Pull request merged**